### PR TITLE
Adding code for publishing to Sonatype.

### PR DESCRIPTION
**Pull request by pwendell**
By default - I'm leaving this commented out. This is because
there is a bug in the PGP signing plugin which causes it to active
even duing a publish-local. So we'll just uncomment when we decide
to publish.

### CODE REVIEW

File: project/SparkBuild.scala [hunk 1]
```diff
@@ -33,7 +35,51 @@ object SparkBuild extends Build {
...
+      "sonatype-staging" at "https://oss.sonatype.org/service/local/staging/deploy/maven2/"),
+
+    publishMavenStyle := true,
+
+    useGpg in Global := true,
+
+    pomExtra := (
+      <url>http://spark-project.org/</url>
+      <licenses>
+	<license>
```
Comment (mateiz) [inline project/SparkBuild.scala]: Are there tabs here? Looks weird.

Comment (pwendell) [inline project/SparkBuild.scala]: good catch - I fixed with a new commit.

**Comment by mateiz** [comment]
Okay, thanks.

**Comment by mateiz** [comment]
By the way, the same issue seems to affect publish-local on branch 0.5. I'll fix it there too.

**Pull request merged**